### PR TITLE
Added coverage for cache middleware

### PR DESCRIPTION
**Pull request by jmauerhan**
Fixes Issue #31

This is a bit of a weird test (which might explain why this hadn't been tackled before), but I think I explained it okay :)  Hope this helps :)

By the way, I didn't notice you were using Mockery until after I did this - the alternative way to test this would be using Mockery and using a mock of the HandlerStack. If you'd prefer that, just let me know - I'd be happy to redo this.

### CODE REVIEW

File: tests/unit/RestClientTest.php [hunk 1]
```diff
@@ -98,4 +99,25 @@ public function testSetAuthMethod()
...
+        $stackProperty->setAccessible(true);
+        $stack = $stackProperty->getValue($handlerStack);
+        //Look for the middleware
+        foreach ($stack as $stackItem) {
+            if ($stackItem[1] === 'snorlax-cache' && $stackItem[0] instanceof CacheMiddleware) {
+                return true;
+            }
+        }
+        //Middleware does not exist on the stack
+        $this->assertTrue(false, 'No CacheMiddleware named snorlax-cache was found');
```
Comment (pedrochaves) [inline tests/unit/RestClientTest.php]: I believe PHPUnit has a `$this->fail('No CacheMiddleware named snorlax-cache was found');` method.

Can you change it?

**Comment by pedrochaves** [comment]
Nice approach to this test :)

About using `Mockery`, we're thinking about refactoring all the tests to use `Prophecy`, since it comes out of the box with PHPUnit 5.4 and I believe it's a little bit easier to work with. Less libraries and less overhead on our tests :)

File: tests/unit/RestClientTest.php (hunk 1 above)
Comment (jmauerhan) [inline tests/unit/RestClientTest.php]: I knew that existed, just had a brain fart :) Changing it now!

Comment (pedrochaves) [inline tests/unit/RestClientTest.php]: Nice! Thank you :D

**Pull request merged**